src/plan_computation.py

```diff
@@ -22,8 +22,41 @@ def compute_distance_between_images(
     Returns:
         The horizontal and vertical distance between images (as a 2-element array).
     """
-    raise NotImplementedError()
+    # Validate overlap and sidelap
+    overlap = float(dataset_spec.overlap)
+    sidelap = float(dataset_spec.sidelap)
+
+    if not (0.0 <= overlap < 1.0):
+        raise ValueError(f"overlap must be in [0, 1), got {overlap}")
+    if not (0.0 <= sidelap < 1.0):
+        raise ValueError(f"sidelap must be in [0, 1), got {sidelap}")
+
+    # Compute the footprint of a single image on the surface at the height
+    footprint = compute_image_footprint_on_surface(camera, dataset_spec.height)
+    footprint_x, footprint_y = float(footprint[0]), float(footprint[1])
+
+    # Distance between image centers
+    distance_x = footprint_x * (1.0 - overlap)
+    distance_y = footprint_y * (1.0 - sidelap)
+
+    return np.array([distance_x, distance_y], dtype=float)
+
+def compute_non_nadir_footprint(camera, height, camera_angle_rad):
```

Comment:
This is an interesting implementation. I would have intuitively thought that I am changing the distance to surface for the same math, so that height becomes height/cos(angle). But you apply the adjustment directly to the footprint. Nice work!